### PR TITLE
Consolidate release candidate changelogs for v0.35.

### PR DESCRIPTION
**Pull request by creachadair**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -63,6 +43,12 @@ Special thanks to external contributors on this release: @JayT106, @bipulprasad,
   - [abci/counter] [\#6684](https://github.com/tendermint/tendermint/pull/6684) Delete counter example app
 
 - Go API
+  - [crypto/armor]: [\#6963](https://github.com/tendermint/tendermint/pull/6963) remove package which is unused, and based on
+    deprecated fundamentals. Downstream users should maintain this
+    library. (@tychoish)
+  - [state] [store] [proxy] [rpc/core]: [\#6937](https://github.com/tendermint/tendermint/pull/6937) move packages to
```
Comment (williambanfield) [inline CHANGELOG.md:49]: Consolidate with line 55.

Comment (creachadair) [reply CHANGELOG.md:49]: Oops, sorry, I missed this comment. It's a good idea, though, I'll do that separately.

**Comment by cmwaters** [comment]
I think historically we haven't squashed release candidate changelogs until the final release just so people know when they bump the release candidates what changes have been made in between them. I was under the impression we were going to cut `-rc3` first.

**Comment by creachadair** [comment]
> I think historically we haven't squashed release candidate changelogs until the final release just so people know when they bump the release candidates what changes have been made in between them. I was under the impression we were going to cut `-rc3` first.

I can revert this. We weren't intending on an RC3 yet when I made the PR. 🙂

**Comment by creachadair** [comment]
See #7061.

**Pull request merged**